chatbot-frontend/component.js

```diff
@@ -213,14 +214,72 @@
 
             chatbox.removeChild(loadingContainer);
 
-            const botMessageHtml = data.message ? DOMPurify.sanitize(data.message) : data.error;
+            const botMessageHtml = data.message ? DOMPurify.sanitize(data.message) : 'error: ' + data.error;
             let isError = false;
             if (!data.message) {
                 console.error('Error from chatbot API:', data.error);
                 isError = true;
             }
             appendMessage(chatbox, botMessageHtml, `${baseUrl}/chatbot/api/frontend/chatbot-logo.png`, false, isError);
             chatId = data.chatId;
+
+            if (!isError) {
```

Comment:
To decrease front end code size, security, and number of API calls needed the conversation storage logic should be moved to the back end, specifically the /chat/:chatbotId route should handle storing conversation history.